File: config/channels.js
// Every feature that posts to a specific channel reads from here instead
// of hardcoding an ID inline. Add a new env var + a new key here whenever
// a new feature needs its own channel (e.g. LEVEL_UP_CHANNEL_ID).
export const CHANNELS = {
  welcome: process.env.WELCOME_CHANNEL_ID,
  roleSelect: process.env.ROLE_SELECT_CHANNEL_ID,
  announcements: process.env.ANNOUNCEMENTS_CHANNEL_ID,
  modLog: process.env.MOD_LOG_CHANNEL_ID,
};

// Small helper so callers don't repeat the "fetch by id" boilerplate
// or crash with an unhelpful error if a channel ID was never set.
export async function getChannel(client, key) {
  const id = CHANNELS[key];
  if (!id) {
    console.warn(`No channel ID configured for "${key}" — check your .env`);
    return null;
  }
  return client.channels.fetch(id);
}

File: config/channels.py
"""Channel configuration.

Every feature that posts to a specific channel reads from here instead of
hardcoding an ID inline. Add a new env var + a new key whenever a feature
needs its own channel.
"""

from __future__ import annotations

import os

import discord


CHANNELS = {
    "welcome": os.getenv("WELCOME_CHANNEL_ID"),
    "roleSelect": os.getenv("ROLE_SELECT_CHANNEL_ID"),
    "announcements": os.getenv("ANNOUNCEMENTS_CHANNEL_ID"),
    "modLog": os.getenv("MOD_LOG_CHANNEL_ID"),
    "kiosk": os.getenv("KIOSK_CHANNEL_ID"),
}

SUPPORT_CATEGORY_ID = os.getenv("SUPPORT_CATEGORY_ID")


async def get_channel(client: discord.Client | discord.Bot, key: str) -> discord.TextChannel | None:
    """Fetch one of the configured channels, logging a warning if unset."""
    channel_id = CHANNELS.get(key)
    if not channel_id:
        print(f'[channels] No channel ID configured for "{key}" - check your .env')
        return None
    try:
        return await client.fetch_channel(channel_id)
    except discord.DiscordException as err:
        print(f'[channels] Could not fetch "{key}" ({channel_id}): {err}')
        return None